drivers/led/lp50xx.c

```diff
@@ -0,0 +1,308 @@
+/*
+ * Copyright (c) 2020 Seagate Technology LLC
+ * Copyright (c) 2022 Grinn
+ *
+ * SPDX-License-Identifier: Apache-2.0
+ */
+
+#define DT_DRV_COMPAT ti_lp50xx
+
+/**
+ * @file
+ * @brief LP50xx LED controller
+ */
+
+#include <zephyr/drivers/gpio.h>
+#include <zephyr/drivers/i2c.h>
+#include <zephyr/drivers/led.h>
+#include <zephyr/drivers/led/lp503x.h>
+#include <zephyr/device.h>
+#include <zephyr/zephyr.h>
+
+#define LOG_LEVEL CONFIG_LED_LOG_LEVEL
+#include <zephyr/logging/log.h>
+LOG_MODULE_REGISTER(lp50xx);
+
+#define LP50XX_DEVICE_CONFIG0		0
+#define   CONFIG0_CHIP_EN		BIT(6)
+
+#define LP50XX_DEVICE_CONFIG1		0x1
+#define   CONFIG1_LED_GLOBAL_OFF	BIT(0)
+#define   CONFIG1_MAX_CURRENT_OPT	BIT(1)
+#define   CONFIG1_PWM_DITHERING_EN	BIT(2)
+#define   CONFIG1_AUTO_INCR_EN		BIT(3)
+#define   CONFIG1_POWER_SAVE_EN		BIT(4)
+#define   CONFIG1_LOG_SCALE_EN		BIT(5)
+
+#ifdef CONFIG_LP503X
```

Comment:
Have you tried to build both drivers? It won't be feasible to use both drivers simultaneously if LP503X and  LP5009_12 variants are mounted on the board.